src/redis-cli.c

```diff
@@ -2336,6 +2339,29 @@ static clusterManagerNode *clusterManagerNewNode(char *ip, int port) {
     return node;
 }
 
+static clusterManagerNode *clusterManagerNewNodeByAddr(char *addr) {
+    char *c = strrchr(addr, '@');
+    int bus_port = -1;
+    if (c != NULL) {
+        bus_port = atoi(c+1);
+        *c = '\0';
+    }
+    c = strrchr(addr, ':');
+    if (c == NULL) {
+        fprintf(stderr, "Invalid address format: %s\n", addr);
+        return 0;
```

Comment:
Better to use `NULL` here instead of 0. It's more conventional and it results in better code readability.